File: js/1-9/problem6.js
const prototypes = require('../prototypes');

let sumOfSquares = function(max) {
	var sum = 0;
	for (let i = 1; i <= max; i++) {
		sum += Math.pow(i, 2);
	}
	return sum;
};

let squareOfTheSum = function(max) {
	var sum = 0;
	for (let i = 1; i <= max; i++) {
		sum += i;
	}
	return Math.pow(sum, 2);
};

// Finds the difference between a and b
let difference = function(a, b) {
	return (a > b) ? a - b : b - a;
};

console.log(difference(squareOfTheSum(100), sumOfSquares(100)));
console.log(difference(sumOfSquares(100), squareOfTheSum(100)));

File: python/1-9/problem6.py
def sum_square_difference(n):

    # Sum of numbers 1 squared to 'n' squared.
    sum_of_squares = sum([x ** 2 for x in range(1, n + 1)])

    # Sum for all numbers from 1 to 'n' then squares that number.
    square_of_the_sum = sum([x for x in range(1, n + 1)]) ** 2

    return abs(sum_of_squares - square_of_the_sum)


if __name__ == '__main__':
    print(sum_square_difference(100))
